### PR TITLE
Add JSX support to DeckGL component and remove jsx-layers example

### PR DESCRIPTION
**Pull request by ibgreen**

### CODE REVIEW

File: docs/get-started/using-with-react.md [hunk 1]
```diff
@@ -1,20 +1,18 @@
 # Using deck.gl with React
 
-deck.gl works perfectly for [React](https://facebook.github.io/react/)-based applications. All deck.gl layers fit naturally into React's component render flow.
+While not directly based on React, deck.gl is designed from ground up to work with [React](https://facebook.github.io/react/)-based applications. deck.gl layers fit naturally into React's component render flow and flux/redux based appications. deck.gl layers will be performantly rerendered whenever you rerender your normal JSX or React components.
```
Comment (balthazar) [inline docs/get-started/using-with-react.md]: The link to the react doc should be put to the first occurrence of React, but I'm sure people that go that far in the docs know what React is anyway

File: docs/get-started/using-with-react.md [hunk 2]
```diff
@@ -32,9 +30,33 @@ const data = [
...
+An important companion to deck.gl is `react-map-gl`. It can provide both a base map as well as event handling for deck.gl.
+
+```jsx
+/// app.js
+import MapGL from 'react-map-gl';
+
+// Set your mapbox access token here
+const MAPBOX_ACCESS_TOKEN = 'MAPBOX_ACCESS_TOKEN';
+
+... // as in previous example
```
Comment (balthazar) [inline docs/get-started/using-with-react.md]: not sure if these two comments are easily understandable / necessary

File: docs/get-started/using-with-react.md [hunk 3]
```diff
@@ -47,7 +69,23 @@ export default class App extends Component {
...
+  render() {
+    return (
+      <DeckGL {...viewport}>
+        <LineLayer id="line-layer" data={data} />
+      <DeckGL />
+    );
+  }
+```
+
+> Caveat: The JSX layer syntax is limitated in that it only works when the layers are direct children of the `DeckGL` component. deck.gl layers are not true React components and cannot be rendered independently by React, and the JSX support depends on deck.gl intercepting the JSX generated child elements before React tries to render them.
```
Comment (balthazar) [inline docs/get-started/using-with-react.md]: maybe put a little `:warning:` emoji in front on "Caveat" for more attention

File: docs/get-started/using-with-react.md (hunk 2 above)
Comment (ibgreen) [reply docs/get-started/using-with-react.md]: Do you suggest we duplicate the entire thing?

File: docs/get-started/using-with-react.md (hunk 3 above)
Comment (ibgreen) [reply docs/get-started/using-with-react.md]: didn't know emoji are supported in markdown. Does that work with both github docs and ocular?

Comment (balthazar) [reply docs/get-started/using-with-react.md]: Definitely works in github, just need a [little tweak](https://github.com/chjj/marked/issues/233#issuecomment-323943368) for support in ocular, but deck.gl is still not ocularized right?

Comment (ibgreen) [reply docs/get-started/using-with-react.md]: True, we'd need to update the deck.gl website code. Probably not worth it right now.

File: docs/get-started/using-with-react.md (hunk 2 above)
Comment (balthazar) [reply docs/get-started/using-with-react.md]: Either duplicate content, or omit obvious things like the viewport object that have already been explained and the react/deck.gl imports. Also maybe change the default mapbox token string to something like `pk.`, so it makes it more obvious. But that's just a thought, don't feel pressure to change it

**Comment by ibgreen** [comment]
Merging per today's discussion

**Pull request merged**